### PR TITLE
[release/5.0] Fix CompGov bugs

### PR DESCRIPTION
**Pull request by wtgodbe**
Fix (at least) most of the Medium+ release/5.0 CompGov bugs

### CODE REVIEW

**Comment by wtgodbe** [comment]
```
$ webpack --mode production --env.production --env.configuration=Release
  internal/modules/cjs/loader.js:1102
        throw new ERR_REQUIRE_ESM(filename, parentPath, packageJsonPath);
        ^
  
  Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: D:\a\_work\1\s\src\Components\WebAssembly\WebAssembly.Authentication\src\Interop\node_modules\ansi-regex\index.js
  require() of ES modules is not supported.
  require() of D:\a\_work\1\s\src\Components\WebAssembly\WebAssembly.Authentication\src\Interop\node_modules\ansi-regex\index.js from D:\a\_work\1\s\src\Components\WebAssembly\WebAssembly.Authentication\src\Interop\node_modules\strip-ansi\index.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
  Instead rename D:\a\_work\1\s\src\Components\WebAssembly\WebAssembly.Authentication\src\Interop\node_modules\ansi-regex\index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from D:\a\_work\1\s\src\Components\WebAssembly\WebAssembly.Authentication\src\Interop\node_modules\ansi-regex\package.json.
  
      at new NodeError (internal/errors.js:322:7)
      at Object.Module._extensions..js (internal/modules/cjs/loader.js:1102:13)
      at Module.load (internal/modules/cjs/loader.js:950:32)
      at Function.Module._load (internal/modules/cjs/loader.js:790:12)
      at Module.require (internal/modules/cjs/loader.js:974:19)
      at require (D:\a\_work\1\s\src\Components\WebAssembly\WebAssembly.Authentication\src\Interop\node_modules\v8-compile-cache\v8-compile-cache.js:161:20)
      at Object.<anonymous> (D:\a\_work\1\s\src\Components\WebAssembly\WebAssembly.Authentication\src\Interop\node_modules\strip-ansi\index.js:2:19)
      at Module._compile (D:\a\_work\1\s\src\Components\WebAssembly\WebAssembly.Authentication\src\Interop\node_modules\v8-compile-cache\v8-compile-cache.js:192:30)
      at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
      at Module.load (internal/modules/cjs/loader.js:950:32) {
    code: 'ERR_REQUIRE_ESM'
  }
  error Command failed with exit code 1.
  info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

@javiercn any idea on what to do about this one? Did you ever hit something similar upgrading components in 6.0/main?

**Comment by wtgodbe** [comment]
@javiercn any thoughts on the above?

**Comment by javiercn** [comment]
@wtgodbe what's the process we are following to update these dependencies? Seems that several of them are jumping across major versions and that's very scary. I don't think there's anything we can do here since those packages are consumed transitively by other packages. I don't believe that whatever tool we are using here is producing the expected results.

**Comment by wtgodbe** [comment]
> what's the process we are following to update these dependencies? Seems that several of them are jumping across major versions and that's very scary. I don't think there's anything we can do here since those packages are consumed transitively by other packages. I don't believe that whatever tool we are using here is producing the expected results.

I mostly ran `yarn-audit-fix` in this branch, which is supposed to be the equivalent of `npm audit fix`. Is it obvious to you what components are being upgraded too far? I could manually resolve those down. I did it this way because the volume of alerts makes it very tedious to manually resolve every bug.

**Comment by javiercn** [comment]
> I mostly ran `yarn-audit-fix` in this branch, which is supposed to be the equivalent of `npm audit fix`.

It is "equivalent" but neither of them work proper in my experience. did you happen to use the `--force` parameter?

I've seen npm audit do things like `downgrade` the version of react-scripts to a super ancient version because there was a dependency "with a vulnerability".

**Comment by javiercn** [comment]
Stuff like this is scary. Its updating from 4.1 to 6.0 which sounds like a recipe for disaster
```
ansi-regex@>=4.1.1, ansi-regex@^3.0.0, ansi-regex@^4.0.0, ansi-regex@^4.1.0:
  version "6.0.1"
  resolved "https://registry.yarnpkg.com/ansi-regex/-/ansi-regex-6.0.1.tgz#3183e38fae9a65d7cb5e53945cd5897d0260a06a"
  integrity sha512-n5M855fKb2SsfMIiFFoVrABHJC8QtHwVx+mHWP3QcEqBHYienj5dHSgjbxtC0WEZXYt4wcD6zrQElDPhFuZgfA==
```

**Comment by wtgodbe** [comment]
> It is "equivalent" but neither of them work proper in my experience. did you happen to use the --force parameter?

No, I didn't. If that means we're stuck applying manual resolutions, I'll get going on that.

**Comment by wtgodbe** [comment]
Actually, wait, in this branch I _did_ resolve everything manually - it was only in https://github.com/dotnet/aspnetcore/pull/41160 that I ran `yarn-audit-fix`

File: src/SignalR/clients/ts/FunctionalTests/yarn.lock
Comment (dougbu) [inline src/SignalR/clients/ts/FunctionalTests/yarn.lock:96]: Yet-another major version bump

File: src/SignalR/clients/ts/FunctionalTests/yarn.lock
Comment (dougbu) [inline src/SignalR/clients/ts/FunctionalTests/yarn.lock:1260]: Safe❔

File: eng/Versions.props
Comment (wtgodbe) [reply eng/Versions.props:199]: This is just to resolve a conflict in `src/Servers/IIS/IIS/samples/NativeIISSample/NativeIISSample.csproj`

Comment (wtgodbe) [reply eng/Versions.props:199]: It's also the latest version: https://www.nuget.org/packages/Microsoft.Win32.Primitives/

File: src/Components/Web.JS/yarn.lock
Comment (wtgodbe) [reply src/Components/Web.JS/yarn.lock:5320]: I don't think we need to worry about these as long as the tests pass, @javiercn @TanayParikh @MackinnonBuck do you agree?

File: src/SignalR/clients/ts/FunctionalTests/yarn.lock
Comment (wtgodbe) [reply src/SignalR/clients/ts/FunctionalTests/yarn.lock:1260]: Are you asking about the bump from 0.0.1? 1.2.6 is the minimum version in any band that satisfies CG, so we don't have much of a choice. @BrennanConroy are you concerned about this update?

Comment (BrennanConroy) [reply src/SignalR/clients/ts/FunctionalTests/yarn.lock:1260]: It is rare to see these small dependency updates break anything and it's especially not a concern for test code. Since if it breaks we'll know about it 😆

**Review by dougbu — APPROVED** [review]
Works for me but please get Blazor team sign-off on some of the bigger `npm` version bumps

File: src/Components/Web.JS/yarn.lock
Comment (wtgodbe) [reply src/Components/Web.JS/yarn.lock:5320]: @javiercn @TanayParikh @MackinnonBuck any thoughts on this update?

Comment (MackinnonBuck) [reply src/Components/Web.JS/yarn.lock:5320]: It seems reasonable to me that if the tests pass, it's probably a safe update. Might be best to wait for Javier's response just to be sure.

Comment (javiercn) [reply src/Components/Web.JS/yarn.lock:5320]: Well, if the tests pass... I don't have a concrete objection.

**Pull request merged**
